app/models/site_upload.rb

```diff
@@ -19,7 +19,15 @@
 class SiteUpload < ApplicationRecord
   include Attachmentable
 
+  FAVICON_SIZES = [16, 32, 48].freeze
+  APPLE_ICON_SIZES   = [57, 60, 72, 76, 114, 120, 144, 152, 167, 180, 1024].freeze
+  ANDROID_ICON_SIZES = [36, 48, 72, 96, 144, 192, 256, 384, 512].freeze
+
+  APP_ICON_SIZES = (APPLE_ICON_SIZES + ANDROID_ICON_SIZES).uniq.freeze
+
   STYLES = {
+    app_icon: APP_ICON_SIZES.each_with_object({}) { |size, hash| hash[size.to_s.to_sym] = "#{size}x#{size}#" }.freeze,
+    favicon: FAVICON_SIZES.each_with_object({}) { |size, hash| hash[size.to_s.to_sym] = "#{size}x#{size}#" }.freeze,
```

Comment:
These graphics will be generated for every upload, e.g. server thumbnail. I think there should be a way to choose the appropriate styles for only specific `var` values.

Reply:
Correct me if I'm wrong, From what I understand, you're saying the current implementation generates graphics (icons and favicons) for every upload of an image, even server thumbnail?

That doesn't seem to be the case. I uploaded a server thumbnail and tested it with the with the sizes for favicon and app icons and i don't see them being rendered. Am I missing something?

e.g.

**server thumbnail with "thumbnail" styles:**
```haml
image_tag @admin_settings.thumbnail.file.url('@2x'), class: 'fields-group__thumbnail'
```
<img width="787" alt="Screenshot 2024-05-04 at 5 19 38 PM" src="https://github.com/mastodon/mastodon/assets/17722450/26bc2635-5d1c-45a8-8c70-d9f660e5697f">


**server thumbnail with "favicon" (or app icon styles):**
```haml
image_tag @admin_settings.thumbnail.file.url('48'), class: 'fields-group__thumbnail'
```
<img width="801" alt="Screenshot 2024-05-04 at 5 20 05 PM" src="https://github.com/mastodon/mastodon/assets/17722450/2a8e444f-90d0-40f5-9702-553b7e9792ec">

Reply:
I think @Gargron is wrong here, the `STYLES` constant does not define styles that apply to each kind of upload, but defines per-var styles already (through `STYLES[file.instance.var.to_sym]`):
https://github.com/mastodon/mastodon/blob/737ddd67f2ed7c0cc5ac342d9547a85f387977e6/app/models/site_upload.rb#L52